docs/source/tutorials/gateware_blinky.rst

```diff
@@ -0,0 +1,177 @@
+
+
+Create a new Amaranth module
+============================
+
+Create a new file called ``gateware-blinky.py`` and add the following code to it:
+
+.. code-block :: python
+    :linenos:
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            return m
+
+Amaranth designs are built from a hierarchy of smaller modules, which are called `elaboratables`. The ``Top`` class expresses that this will be the top-level or entry-point of your design.
+
+Right now the ``Top`` module does not do anything except to create an Amaranth ``Module()`` and return it from the ``elaborate(...)`` method.
+
+The ``elaborate(...)`` method also takes an argument called ``platform`` that contains resources specific to the board or platform the module is compiled for.
+
+In this case, the argument will be an instance of the `Cynthion Board Description <https://github.com/greatscottgadgets/cynthion/blob/main/cynthion/python/src/gateware/platform/cynthion_r1_4.py>`__ and contain a map of Cynthion resources such as LEDs, USB PHY's, USER PMOD connectors and board constraints.
+
+
+Obtain a platform resource
+==========================
+
+Edit ``gateware-blinky.py`` and add the highlighted line:
+
+.. code-block :: python
+    :linenos:
+    :emphasize-lines: 7
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            leds: Signal(6) = Cat(platform.request("led", n).o for n in range(0, 6))
+
+            return m
+
+Amaranth platform resources can be obtained via the ``platform.request(name, number=0)`` method where ``name`` is the name of the resource and ``number`` is the index of the resource if there are more than one defined.
+
+In this case we use a Python list comprehension to obtain all six FPGA led's and concatenate them into a six-bit addressable Amaranth ``Signal`` using the ``Cat`` operation.
+
+
+Timer State
+===========
+
+To make the LED blink at predictable intervals we'll use a simple timer.
+
+To start with, let's define the timer state by adding the highlighted lines:
+
+.. code-block :: python
+    :linenos:
+    :emphasize-lines: 9-10
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            leds: Signal(6) = Cat(platform.request("led", n).o for n in range(0, 6))
+
+            half_freq: int    = int(60e6 // 2)
+            timer: Signal(25) = Signal(range(half_freq))
+
+            return m
+
+First we'll declare a variable ``half_freq`` which is exactly half of Cynthion FPGA's default clock frequency in Hz, next we'll declare ``timer`` to be an Amaranth ``Signal`` which is wide enough to contain a value equal to ``half_freq``.
+
+If we increment the ``timer`` by one for each clock cycle until it reaches ``half_freq`` we get a timer with a 500ms period.
```

Comment:
Let's change this to be:
- "... wide enough to contain a value equal to `half_freq - 1`"
- "... until it reaches `half_freq - 1`"

Reply:
👌 Fixed in: https://github.com/greatscottgadgets/cynthion/pull/197/commits/c61b562814199464469195ea181a1bec8f411cb6